test/libsolidity/AnalysisFramework.h

```diff
@@ -35,8 +35,8 @@ namespace solidity
 
 class Type;
 class FunctionType;
-using TypePointer = std::shared_ptr<Type const>;
-using FunctionTypePointer = std::shared_ptr<FunctionType const>;
+using TypePointer = Type const*;
```

Comment:
This is a critical change. It changes a non-POD to a POD type, which means that local variables of TypePointer type are no longer guaranteed to be zero-initialized. Did you check if that can lead to problems?

Reply:
The compiler actually told me all those problems, tbh. After I found out that the compiler told me some spots I started searching with REs to all `TypePointer` variable declarations and made sure they're `nullptr` initialized if not explicitly initialized right after.